### PR TITLE
feat: create a delete all testimonials endpoint 

### PR DESCRIPTION
**Pull request by yvonnegichovi**
## Description
This PR introduces the following changes:

- Fixed DELETE Endpoint: Improved the DELETE endpoint at `/testimonials/` to correctly delete all testimonial entries from the database.
- Pytest: Created pytest to test the newly added endpoint.

**Related Issue**
#270 [FEAT]: API Endpoint to Delete All Testimonials

**Motivation and Context**
- The DELETE endpoint provides a way to clear all testimonials, which is useful for maintenance or testing purposes. Fixing the model warnings ensures that pytest can correctly collect and test the model without issues.

**How Has This Been Tested?**
- **Manual Testing:**
-Tested the DELETE endpoint to ensure it correctly removes all testimonials from the database.

-**Automated Testing:**

- Updated and ran unit tests using pytest to ensure all new and existing tests pass.
- Validated that the Testimonial model no longer triggers pytest warnings.

**The endpoint**
- Route `DELETE api/v1/testimonials`
- Method: DELETE
- Parameters: None

**DELETE Request Example:**

- Request URL: http://localhost:8000/testimonials/
- Response: {"message": "All testimonials deleted successfully", "data": {}, "status_code": 200}

**Types of changes**
- [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**
- [x]  My code follows the code style of this project.
 - [x] My change requires a change to the documentation.
- [x]  I have updated the documentation accordingly.
 - [x] I have read the CONTRIBUTING document.
 - [x] I have added tests to cover my changes.
 - [x] All new and existing tests passed.

### CODE REVIEW

File: .replit [hunk 1]
```diff
@@ -0,0 +1,8 @@
+modules = ["python-3.12", "python-3.10"]
+run = "uvicorn main:app --host 0.0.0.0 --port 3000"
+
+[nix]
+channel = "stable-24_05"
+
+[deployment]
+run = ["sh", "-c", "uvicorn main:app --host 0.0.0.0 --port 3000"]
```
Comment (Goketech) [inline .replit:8]: remove this file

File: api/v1/models/testimonial.py [hunk 2]
```diff
@@ -12,4 +12,4 @@ class Testimonial(BaseTableModel):
     content = Column(Text, nullable=False)
     ratings = Column(Float, nullable=True)
 
-    author = relationship("User", back_populates="testimonials")
\ No newline at end of file
+    author = relationship("User", back_populates="testimonials")
```
Comment (Goketech) [inline api/v1/models/testimonial.py:15]: revert back the empty line

File: api/v1/routes/testimonial.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+from fastapi import Depends, HTTPException, APIRouter, Request, Response, status
+from sqlalchemy.orm import Session
+from fastapi.responses import JSONResponse
+
+from api.utils.success_response import success_response
+from api.v1.models.testimonial import Testimonial
+from api.v1.models.user import User
+# from api.v1.schemas.user import DeactivateUserSchema, UserBase
```
Comment (Goketech) [inline api/v1/routes/testimonial.py:8]: remove this comment

File: main.py [hunk 4]
```diff
@@ -16,7 +16,6 @@
 )
 from api.v1.routes import api_version_one
 
-
```
Comment (Goketech) [inline main.py:19]: revert this

File: requirements.txt [hunk 5]
```diff
@@ -1,5 +1,4 @@
 pytest-mock==3.14.0
-
```
Comment (Goketech) [inline requirements.txt:2]: revert this

File: api/v1/routes/testimonial.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            message="All testimonials deleted successfully",
+            data={},
+            status_code=status.HTTP_200_OK
+        )
+    except Exception as e:
+        db.rollback()
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=str(e)
+        )
```
Comment (Goketech) [inline api/v1/routes/testimonial.py:65]: do not make db queries in the routes folder, create a testimonial.py file in service and make your queries there and call the function here

**Review by Goketech — CHANGES_REQUESTED** [review]

**Review by yvonnegichovi — COMMENTED** [review]
Changes made as requested

**Review by yvonnegichovi — COMMENTED** [review]
Changes reviewed

**Comment by Goketech** [comment]
resolve the conversations

**Review by zxenonx — APPROVED** [review]
Thanks @yvonnegichovi

**Pull request merged**